### PR TITLE
Am 1115 pact consumer

### PR DESCRIPTION
**Pull request by isaacola-hub**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/AM-1115

### Change description ###

Added three additional consumer PACT for create role assignment endpoint interaction from ORM to RAS.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTest.java [hunk 1]
```diff
@@ -66,6 +69,54 @@ public RequestResponsePact executeGetListOfRolesAndGet200(PactDslWithProvider bu
...
+                .headers(getRoleAssignmentResponseHeaders())
+                .body(createRoleAssignmentResponseReplaceExistingFalse())
+                .toPact();
+    }
+
+    @Pact(provider = "am_role_assignment_service", consumer = "am_org_role_mapping")
+    public RequestResponsePact executeCreateRoleAssignmentOneRoleAndGet201(PactDslWithProvider builder) {
+
+        return builder
+                .given("The assignment request is valid with one requested role")
```
Comment (nitinsahu1) [inline src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTest.java:92]: We can provide more details here about this test 
e.g. "The assignment request is valid with one requested role and replaceExisting flag as true"

File: src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTest.java [hunk 2]
```diff
@@ -66,6 +69,54 @@ public RequestResponsePact executeGetListOfRolesAndGet200(PactDslWithProvider bu
...
+                .headers(getRoleAssignmentResponseHeaders())
+                .body(createRoleAssignmentResponseOneRole())
+                .toPact();
+    }
+
+    @Pact(provider = "am_role_assignment_service", consumer = "am_org_role_mapping")
+    public RequestResponsePact executeCreateRoleAssignmentZeroRoleAndGet201(PactDslWithProvider builder) {
+
+        return builder
+                .given("The assignment request is valid with zero requested role")
```
Comment (nitinsahu1) [inline src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTest.java:108]: We can provide more details here about this test
e.g. "The assignment request is valid with zero requested role and replaceExisting flag as true"

File: src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTest.java [hunk 3]
```diff
@@ -93,6 +151,96 @@ void getListOfRolesAndGet200Test(MockServer mockServer)
...
+
+        JSONObject response =  new JSONObject(actualResponseBody);
+        Assertions.assertThat(response).isNotNull();
+
+        JSONObject roleRequest = response.getJSONObject("roleAssignmentResponse").getJSONObject("roleRequest");
+        JSONArray requestedRoles = response.getJSONObject("roleAssignmentResponse").getJSONArray("requestedRoles");
+        JSONObject role1 = requestedRoles.getJSONObject(0);
+        assertThat(roleRequest.get("status"), equalTo("APPROVED"));
+        assertThat(roleRequest.get("requestType"), equalTo("CREATE"));
+        assertThat(roleRequest.get("process"), equalTo(roleRequest.get("reference")));
```
Comment (nitinsahu1) [inline src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTest.java:173]: It is not mandatory to have both process and reference equal.
We can assert them separately.
assertThat(roleRequest.get("process"), equalTo("S-028"));
assertThat(roleRequest.get("reference"), equalTo("S-028"));

File: src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTest.java (hunk 2 above)
Comment (isaacola-hub) [reply src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTest.java]: Done

File: src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTest.java (hunk 1 above)
Comment (isaacola-hub) [reply src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTest.java]: Done

File: src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTest.java (hunk 3 above)
Comment (isaacola-hub) [reply src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTest.java]: Done

**Review by nitinsahu1 — APPROVED** [review]

**Review by aashishaggarwal123 — APPROVED** [review]

**Pull request merged**